### PR TITLE
support recordio in dist train

### LINKED ISSUE #11024: Use recordIO in distributed training

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #11024

### CODE REVIEW

File: paddle/fluid/operators/reader/create_recordio_file_reader_op.cc [hunk 1]
```diff
@@ -65,20 +65,22 @@ class CreateRecordIOReaderOp : public framework::OperatorBase {
                       static_cast<int>(shape_concat.size()),
                       "The accumulate of all ranks should be equal to the "
                       "shape concat's length.");
-    std::string filename = Attr<std::string>("filename");
+    auto filenames = Attr<std::vector<std::string>>("filenames");
```
Comment (typhoonzero) [inline paddle/fluid/operators/reader/create_recordio_file_reader_op.cc]: `fluid.layers.open_files` should do the job, please see `test_multi_file_reader.py`

Comment (Yancey1989) [reply paddle/fluid/operators/reader/create_recordio_file_reader_op.cc]: Seems `open_files_op` can only deal with the normal files.

Comment (JiayiFeng) [reply paddle/fluid/operators/reader/create_recordio_file_reader_op.cc]: `open_files_op` can deal with RecordIO file. What's your 'unnormal' file?

File: doc/fluid/howto/cluster/fluid_recordio.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+interface to deal with the RecordIO files.
+
+## Generate RecordIO File
+
+Before start training with RecordIO files, you need to convert your training data
+to RecordIO format by `fluid.recordio_writer.convert_reader_to_recordio_file`, the sample codes
+as follows:
+
+```python
+    reader = paddle.batch(mnist.train(), batch_size=1)
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply doc/fluid/howto/cluster/fluid_recordio.md]: Done.

File: doc/fluid/howto/cluster/fluid_recordio.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        tmp, = exe.run(fetch_list=[avg_loss])
+
+        avg_loss_np.append(tmp)
+        print(batch_id)
+        batch_id += 1
+```
+
+## Use the RecordIO files in Distributed Training
+
+1. generate multiple RecordIO files
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply doc/fluid/howto/cluster/fluid_recordio.md]: Done.

File: doc/fluid/howto/cluster/fluid_recordio.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            fluid.layers.data(
+            name='image', shape=[784]),
+            fluid.layers.data(
+            name='label', shape=[1], dtype='int64'),
+        ],
+        place=fluid.CPUPlace())
+    fluid.recordio_writer.convert_reader_to_recordio_file('./mnist.recordio', reader, feeder)
+```
+
+The above codes would generate a RecordIO `./mnist.recordio` on your host.
```
[the remark this answers is not in the document]
Comment (Yancey1989) [reply doc/fluid/howto/cluster/fluid_recordio.md]: Done.

File: python/paddle/fluid/recordio_writer.py [hunk 5]
```diff
@@ -46,3 +48,36 @@ def convert_reader_to_recordio_file(
...
+                                        max_num_records) as writer:
+                for l in lines:
+                    res = feeder.feed(l)
+                    for each in feed_order:
+                        writer.append_tensor(res[each])
+                    writer.complete_append_tensor()
+                    counter += 1
+                lines = []
+                f_idx += 1
+    return counter
```
Comment (typhoonzero) [inline python/paddle/fluid/recordio_writer.py]: Remove `tools/codestyle/docstring_checker.pyc` file please

Comment (Yancey1989) [reply python/paddle/fluid/recordio_writer.py]: Done, and add it to gitignore.

**Pull request merged**